cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/partitioned/PartitionConsumer.java

```diff
@@ -67,4 +67,20 @@
    * @param succeeded whether or not processing of the specified partitions was successful
    */
   void onFinishWithKeys(List<? extends PartitionKey> partitionKeys, boolean succeeded);
+
+  /**
+   * Returns a list of partitions to the working set, without increasing the number of retries. They are made
+   * available for future processing.
+   *
+   * @param partitions list of partitions to put back
+   */
```

Comment:
you should document what happens if the partitions are not in the working set, or not marked as taken.